### ISSUE TITLE
Tensorflow Serving model is returning different output for the same input,and even negative score.

### ISSUE THREAD

**Issue by firewu** [tensorflow/serving#689]
I had trained and exported a tensorflow serving model which is for chinese text classify.I set the input placeholder as  tf.placeholder(tf.float32, [None, sequence_length, embedding_size], name="input_x"),
and output the top 5 of the predict result.                                                                                                                                                                      
    But I get the different output value and score when I input the same text,and  the scores with the negative value,that makes me quit confuse what is wrong with my model or params .                         
**The output may like the following：**                                                                                                                                                                          
   _outputs {                                                                                                                                                                                                    
  key: "classes"                                                                                                                                                                                                 
  value {                                                                                                                                                                                                        
    dtype: DT_STRING                                                                                                                                                                                             
    tensor_shape {                                                                                                                                                                                               
      dim {                                                                                                                                                                                                      
        size: 1                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
      dim {                                                                                                                                                                                                      
        size: 5                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
    }                                                                                                                                                                                                            
    string_val: "\347\231\275\351\205\222"                                                                                                                                                                       
    string_val: "\346\264\213\351\205\222"                                                                                                                                                                       
    string_val: "\351\224\200\345\224\256\346\234\215\345\212\241"                                                                                                                                               
    string_val: "\350\221\241\350\220\204\351\205\222"                                                                                                                                                           
    string_val: "\351\273\204\351\205\222"                                                                                                                                                                       
  }                                                                                                                                                                                                              
}                                                                                                                                                                                                                
outputs {                                                                                                                                                                                                        
  key: "scores"                                                                                                                                                                                                  
  value {                                                                                                                                                                                                        
    dtype: DT_FLOAT                                                                                                                                                                                              
    tensor_shape {                                                                                                                                                                                               
      dim {                                                                                                                                                                                                      
        size: 1                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
      dim {                                                                                                                                                                                                      
        size: 5                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
    }                                                                                                                                                                                                            
    float_val: 56.5367240906                                                                                                                                                                                     
    float_val: -3.92663836479                                                                                                                                                                                    
    float_val: -6.48062610626                                                                                                                                                                                    
    float_val: -17.449054718                                                                                                                                                                                     
    float_val: -32.129032135                                                                                                                                                                                     
  }                                                                                                                                                                                                              
}_                                                                                                                                                                                                               
**anthoer different output：**                                                                                                                                                                                   
outputs {                                                                                                                                                                                                        
  key: "classes"                                                                                                                                                                                                 
  value {                                                                                                                                                                                                        
    dtype: DT_STRING                                                                                                                                                                                             
    tensor_shape {                                                                                                                                                                                               
      dim {                                                                                                                                                                                                      
        size: 1                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
      dim {                                                                                                                                                                                                      
        size: 5                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
    }                                                                                                                                                                                                            
    string_val: "\347\231\275\351\205\222"                                                                                                                                                                       
    string_val: "\351\273\204\351\205\222"                                                                                                                                                                       
    string_val: "\350\221\241\350\220\204\351\205\222"                                                                                                                                                           
    string_val: "\351\224\200\345\224\256\346\234\215\345\212\241"                                                                                                                                               
    string_val: "\346\264\213\351\205\222"                                                                                                                                                                       
  }                                                                                                                                                                                                              
}                                                                                                                                                                                                                
outputs {                                                                                                                                                                                                        
  key: "scores"                                                                                                                                                                                                  
  value {                                                                                                                                                                                                        
    dtype: DT_FLOAT                                                                                                                                                                                              
    tensor_shape {                                                                                                                                                                                               
      dim {                                                                                                                                                                                                      
        size: 1                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
      dim {                                                                                                                                                                                                      
        size: 5                                                                                                                                                                                                  
      }                                                                                                                                                                                                          
    }                                                                                                                                                                                                            
    float_val: 38.6646461487                                                                                                                                                                                     
    float_val: 6.10928010941                                                                                                                                                                                     
    float_val: 2.18265676498                                                                                                                                                                                     
    float_val: -1.94573044777                                                                                                                                                                                    
    float_val: -3.25391077995                                                                                                                                                                                    
  }                                                                                                                                                                                                              
}

**Comment by chrisolston** [comment]
One thing to check: Make sure your exported model doesn't include any queues (e.g. FifoQueueOp [1]). That's a common mistake and it can lead to puzzling and nondeterministic behavior.

[1] https://www.tensorflow.org/api_docs/python/tf/FIFOQueue

**Comment by firewu** [comment]
@chrisolston  Thank you very much.I reviewed my code and I believeed I have find the key to the problem is because the **"dropout=0.5"** is using when predict as I set **dropout** as a constant.
    After I changed it to a placeholder,I found tf serving is not support for multiple input.It raises the error _"[[Node: dropout_keep_prob = Placeholder[dtype=DT_FLOAT, shape=<unknown>, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]")"_
   Finally,I changed **"dropout=1"** as a constant I get the same output ,but it's not a perfect solution to solve it.
   https://github.com/tensorflow/serving/issues/602
   https://github.com/tensorflow/serving/issues/9

**Comment by svpino** [comment]
Using TensorFlow 2.2, when saving a SavedModel that includes Dropouts, TensorFlow serving returns non-deterministic predictions.

How can I disable TF Serving from using the Dropouts layer? Alternatively, how can I remove those Dropout layers from the SavedModel file?
